### PR TITLE
Modify OMIM-AUTO panel to include genes in both builds

### LINKED ISSUE #4070: Option to use different build (hg38) as basis for OMIM-AUTO
As it is a few genes are missing from the OMIM-AUTO updates due to the genes not being annotated on the scout default build (hg37). Lets introduce a build option for the `scout update OMIM` and likely `scout update panelapp-green`.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- New OMIM-AUTO gene panels will contain genes in both builds. This is the easiest way to close #4070

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR, DN

### CODE REVIEW

**Comment by northwestwitch** [comment]
### Main branch:

image

- **4599 genes** (update files downloaded today)

image

### This branch:

image

- **4606 genes**

image

File: scout/adapter/mongo/hgnc.py [hunk 1]
```diff
@@ -336,28 +336,23 @@ def genes_by_alias(self, build="37", genes=None):
...
+            genes_query = {"build": str(build)} if build else {}
+            genes = self.hgnc_collection.find(genes_query)
 
         for gene in genes:
-            # Collect the hgnc_id
             hgnc_id = gene["hgnc_id"]
-            # Collect the true symbol given by hgnc
             hgnc_symbol = gene["hgnc_symbol"]
+
             # Loop aver all aliases
```
Comment (dnil) [inline scout/adapter/mongo/hgnc.py:348]: Good with removing the unneeded comments. That one could also go, but not important.

File: scout/adapter/mongo/hgnc.py [hunk 2]
```diff
@@ -336,28 +336,23 @@ def genes_by_alias(self, build="37", genes=None):
...
             for alias in gene["aliases"]:
-                true_id = None
-                # If the alias is the same as hgnc symbol we know the true id
+                if alias not in alias_genes:
+                    alias_genes[alias] = {"true": None, "ids": set()}
+
+                alias_genes[alias]["ids"].add(hgnc_id)
```
Comment (dnil) [inline scout/adapter/mongo/hgnc.py:353]: ✅ Clearer!

**Review by dnil — APPROVED** [review]
✅

File: scout/adapter/mongo/hgnc.py [hunk 3]
```diff
@@ -336,28 +336,23 @@ def genes_by_alias(self, build="37", genes=None):
         LOG.debug("Fetching all genes by alias")
         # Collect one entry for each alias symbol that exists
         alias_genes = {}
-        # Loop over all genes
-        if not genes:
-            genes = self.hgnc_collection.find({"build": str(build)})
+
+        if genes is None:
+            genes_query = {"build": str(build)} if build else {}
+            genes = self.hgnc_collection.find(genes_query)
```
Comment (dnil) [inline scout/adapter/mongo/hgnc.py:342]: Consider a projection here!

File: scout/adapter/mongo/hgnc.py [hunk 4]
```diff
@@ -339,7 +339,9 @@ def genes_by_alias(self, build=None, genes=None):
 
         if genes is None:
             genes_query = {"build": str(build)} if build else {}
-            genes = self.hgnc_collection.find(genes_query)
+            genes = self.hgnc_collection.find(
+                genes_query, projection={"hgnc_id": 1, "hgnc_symbol": 1, "aliases": 1}
```
Comment (dnil) [inline scout/adapter/mongo/hgnc.py:343]: 🥇

File: scout/adapter/mongo/hgnc.py (hunk 3 above)
Comment (northwestwitch) [reply scout/adapter/mongo/hgnc.py]: Done!

**Pull request merged**